interfaces/kmod/backend.go

```diff
@@ -107,6 +142,11 @@ func (b *Backend) Setup(snapInfo *snap.Info, confinement interfaces.ConfinementO
 func (b *Backend) Remove(snapName string) error {
 	glob := interfaces.SecurityTagGlob(snapName)
 	_, _, err := osutil.EnsureDirState(dirs.SnapKModModulesDir, glob, nil)
+	if err != nil {
+		return err
```

Comment:
Perhaps we should run both operations and then combine both (or just one) error, eg;
```
var errs []error
if _, _, err := osutil.EnsureDirState(dirs.SnapKModModulesDir, glob, nil); err != nil {
    errs = append(errs, err)
}
if _, _, err = osutil.EnsureDirState(dirs.SnapKModModprobeDir, glob, nil); err != nil {
    errs = append(errs, err)
}
if len(errs) > 0 {
    return fmt.Errorf("cannot remove kernel modules config files: %v", errs)
}
```

Reply:
I'd rather wait for more feedback before changing this, because IMHO the current version is both simpler and more readable (despite being more asymmetric).

Reply:
I think I agree with @bboozzoo 's suggestion; makes sense to try to clean as much as possible before reporting the error.

Reply:
Ah, I didn't think of that. Done!